### PR TITLE
Update Ruby and Travis-related config

### PR DESCRIPTION
**Pull request by jgarber623-gov**
This PR reworks the Travis CI configuration as well as setting Ruby version (via `.ruby-version`). Additional comments posted inline.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -4,9 +4,9 @@
...
-end_of_line = lf
-trim_trailing_whitespace = true
 charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+indent_size = 2
 indent_style = space
-indent_size = tab
-tab_width = 4
+trim_trailing_whitespace = true
```
Comment (jgarber623-gov) [inline .editorconfig]: Project files were using two-space indentation already so the existing EditorConfig settings were incorrect.

File: Gemfile [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
-gem 'uswds-jekyll', :git => 'https://github.com/18F/uswds-jekyll.git'
+gem 'html-proofer', '~> 3.8'
+gem 'jekyll', '~> 3.7'
+
+gem 'uswds-jekyll', git: 'https://github.com/18F/uswds-jekyll.git'
+
 group :jekyll_plugins do
-  gem 'jekyll_pages_api_search'
+  gem 'jekyll_pages_api_search', '~> 0.5.0'
 end
```
Comment (jgarber623-gov) [inline Gemfile]: Added some more-specific gem versions to the Gemfile.

File: _config.yml [hunk 3]
```diff
@@ -15,5 +14,13 @@ scripts:
...
+include:
+  - '_pages'
+
+exclude:
+  - 'cibuild'
+  - 'CNAME'
+  - 'Gemfile.lock'
+  - 'Gemfile'
+  - 'README.md'
+  - 'vendor'
```
Comment (jgarber623-gov) [inline _config.yml]: Turns out this YAML syntax works, too. Added `cibuild` to the list of excluded build files and alphabetized the whole thing.

File: .travis.yml [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
+  - 2.5.0
 env:
   global:
     - NOKOGIRI_USE_SYSTEM_LIBRARIES=true # speeds up installation of html-proofer
-
+branches:
+  only:
+    - static-site
+before_install:
+  - gem update --system
```
Comment (jgarber623-gov) [inline .travis.yml]: Updating system gems will resolve [some issues](https://github.com/travis-ci/travis-ci/issues/8978).

File: .editorconfig (hunk 1 above)
Comment (jordangov) [reply .editorconfig]: :+1:

File: _config.yml (hunk 3 above)
Comment (jordangov) [reply _config.yml]: But I like my arrays! :P

Comment (jgarber623-gov) [reply _config.yml]: :expressionless:

File: _config.yml [hunk 5]
```diff
@@ -15,5 +15,13 @@ scripts:
...
 theme: uswds-jekyll
 
-include: ['_pages']
-exclude: ['README.md', 'Gemfile.lock', 'Gemfile', 'CNAME', 'vendor']
+include:
+  - '_pages'
+
+exclude:
+  - 'cibuild'
+  - 'CNAME'
```
Comment (jordangov) [inline _config.yml]: Can you remove this file? My mistake, but we're not using a github CNAME in the new version.

File: .travis.yml [hunk 6]
```diff
@@ -1,18 +1,19 @@
...
+dist: trusty
+sudo: false
 language: ruby
+cache: bundler
 rvm:
-- 2.3.5
-
-script: ./cibuild
-
+  - 2.5.0
```
Comment (nschonni) [inline .travis.yml]: Since you added the ruby-version file, this isn't needed

File: _config.yml (hunk 5 above)
Comment (jgarber623-gov) [reply _config.yml]: Removed!

File: .travis.yml (hunk 6 above)
Comment (jgarber623-gov) [reply .travis.yml]: Removed!

**Comment by jordangov** [comment]
:shipit:

**Pull request merged**